src/System.Reflection.Metadata/src/System/Reflection/Metadata/Decoding/ITypeProvider.cs

```diff
@@ -35,20 +35,5 @@ public interface ITypeProvider<TType>
         /// Note that when the signature comes from a WinMD file additional processing is needed to determine whether the target type is a value type or a reference type.
         /// </param>
         TType GetTypeFromReference(MetadataReader reader, TypeReferenceHandle handle, byte rawTypeKind);
```

Comment:
I'd like to rename ITypeProvider since it only applies to def or ref and isn't used directly, but sounds like the root of the hierarchy. Can't think of shorter name than ITypeDefinitionOrReferenceProvider? Something that conveys non-primitive, but also not constructed/complex. Any ideas?